sources/updater/update_metadata/src/lib.rs

```diff
@@ -320,34 +319,40 @@ impl Update {
         }
     }
 
+    /// Returns whether the update is truly available. A update is said to be 'ready/available' if the wave
+    /// this instance belongs to has fully passed, or if the instance's position in the wave has passed.
+    /// The position of the instance within the wave is determined by the seed value.
     pub fn update_ready(&self, seed: u32) -> bool {
-        // Has this client's wave started
-        if let Some(wave) = self.update_wave(seed) {
-            return wave.has_started();
-        }
-
-        // Or there are no waves
-        true
-    }
-
-    pub fn jitter(&self, seed: u32) -> Option<DateTime<Utc>> {
+        // If this instance is part of some update wave
```

Comment:
If we pass in the time, e.g. `pub fn update_ready(seed, Time::now())`, then the function becomes a ‘pure’ function and the testing would be easier/more-reliable.